src/contexts/Drive/Drive.context.tsx

```diff
@@ -77,6 +82,92 @@ export const DriveContextProvider: React.FC<DriveContextProviderProps> = ({ chil
     return driveUseCases.getFolderContent({ folderId });
   };
 
+  const getModifiedItems = async ({
+    itemsType,
+    limit = 50,
+    offset = 0,
+    updatedAt,
+    status,
+  }: {
+    itemsType: 'files' | 'folders';
+    limit?: number;
+    offset?: number;
+    updatedAt: string;
+    status: 'ALL' | 'TRASHED' | 'REMOVED';
+  }) => {
+    const query = `status=${status}&offset=${offset}&limit=${limit}${updatedAt && `&updatedAt=${updatedAt}`}`;
+    const newToken = await asyncStorageService.getItem(AsyncStorageKey.PhotosToken);
+
+    if (!newToken) return;
+
+    const headers = await getHeaders(newToken);
+
+    try {
+      const modifiedItems = await fetch(`${constants.DRIVE_NEW_API_URL}/${itemsType}?${query}`, {
+        method: 'GET',
+        headers,
+      });
+
+      return modifiedItems;
+    } catch (error) {
+      errorService.reportError(error);
+    }
+  };
+
+  const checkIfItemShouldBeUpdatedOrDeleted = async (folderContentFromDB: FolderContent) => {
+    // 1. Get the modified items from the server
+    const [modifiedFiles, modifiedFolders] = await Promise.all([
+      getModifiedItems({
+        itemsType: 'files',
+        updatedAt: new Date().toISOString(),
+        status: 'ALL',
+      }),
+      getModifiedItems({
+        itemsType: 'files',
+        updatedAt: new Date().toISOString(),
+        status: 'ALL',
+      }),
+    ]);
+
+    const parsedModifiedFiles = await modifiedFiles?.json();
+    const parsedModifiedFolders = await modifiedFolders?.json();
+
+    if (
+      !parsedModifiedFiles ||
+      !parsedModifiedFolders ||
+      folderContentFromDB.children.length === 0 ||
+      folderContentFromDB.files.length === 0
+    ) {
+      return;
+    }
+
+    folderContentFromDB.files.forEach((file: DriveFileData) => {
+      const fileModified = parsedModifiedFiles?.find((item: DriveFileData) => item.id === file.id);
+      if (fileModified) {
+        if (fileModified.status === 'TRASHED' || fileModified.status === 'REMOVED') {
```

Comment:
```
if(file.status === "EXISTS") {
// Item is not deleted
} else {
// Item is deleted
}
```

This is simpler in case you want to simplify this